src/contracts/rewarder/ODefaultOperatorRewards.sol

```diff
@@ -0,0 +1,183 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// Copyright (C) Moondance Labs Ltd.
+// This file is part of Tanssi.
+// Tanssi is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+// Tanssi is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+// You should have received a copy of the GNU General Public License
+// along with Tanssi.  If not, see <http://www.gnu.org/licenses/>
+
+pragma solidity 0.8.25;
+
+import {IODefaultOperatorRewards} from "../../interfaces/rewarder/IODefaultOperatorRewards.sol";
+import {IODefaultStakerRewards} from "../../interfaces/rewarder/IODefaultStakerRewards.sol";
+
+import {INetworkMiddlewareService} from "@symbiotic/interfaces/service/INetworkMiddlewareService.sol";
+import {SimpleKeyRegistry32} from "../libraries/SimpleKeyRegistry32.sol";
+import {MerkleProof} from "@openzeppelin/contracts/utils/cryptography/MerkleProof.sol";
+import {ReentrancyGuard} from "@openzeppelin/contracts/utils/ReentrancyGuard.sol";
+import {SafeERC20, IERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {Math} from "@openzeppelin/contracts/utils/math/Math.sol";
+import {ScaleCodec} from "@snowbridge/src/utils/ScaleCodec.sol";
+
+contract ODefaultOperatorRewards is ReentrancyGuard, IODefaultOperatorRewards {
+    using SafeERC20 for IERC20;
+    using Math for uint256;
+
+    /**
+     * @inheritdoc IODefaultOperatorRewards
+     */
+    address public immutable i_networkMiddlewareService;
+
+    /**
+     * @inheritdoc IODefaultOperatorRewards
+     */
+    address public immutable i_token;
+
+    /**
+     * @inheritdoc IODefaultOperatorRewards
+     */
+    address public immutable i_network;
+
+    /**
+     * @inheritdoc IODefaultOperatorRewards
+     */
+    address public s_defaultStakerRewards;
+
+    /**
+     * @inheritdoc IODefaultOperatorRewards
+     */
+    uint48 public s_operatorShare;
+
+    /**
+     * @inheritdoc IODefaultOperatorRewards
+     */
+    mapping(uint48 epoch => bytes32 value) public s_epochRoot;
+
+    /**
+     * @inheritdoc IODefaultOperatorRewards
+     */
+    mapping(uint48 epoch => BalancePerEpoch balancePerEpoch) public s_balance;
+
+    /**
+     * @inheritdoc IODefaultOperatorRewards
+     */
+    mapping(uint48 epoch => mapping(address account => uint256 amount)) public s_claimed;
```

Comment:
ah I see where you are going. You are considering this epoch to be the network epoch right? and for every "network epoch" you will keep adding when we receive rewards corresponding to that epoch, and then substract when people claim. Am I right here?

Please let me know

Reply:
But this is not going to work I believe. Here is the reason:

- we are sending the rewards with: `total_inflated`, `total_points`, `tanssi_epoch`, `middleware_network_epoch` and `reward_root`. Remember the last 2 can be different, one can be larger than the other for instance. What you know for sure is that the rewards will be sent once per `tanssi_epoch`.

- You dont accumulate rewards per operator when you receive that information. You just get the root. That  means that this variable needs to be a boolean and the epoch has to be the `tanssi_epoch` necessarily